### DISCUSSION TITLE
urequests: 'Response' object has no attribute 'headers'

### DISCUSSION THREAD

**Discussion by gnbl** [micropython/micropython#10788] [ESP32] [answered]
For ESP32 with esp32-[ota-]20220618-v1.19.1.bin on [LILYGO/TTGO T-Display](https://www.lilygo.cc/products/lilygo%C2%AE-ttgo-t-display-1-14-inch-lcd-esp32-control-board) 
> MicroPython v1.19.1 on 2022-06-18; 4MB/OTA module with ESP32
> MicroPython v1.19.1 on 2022-06-18; ESP32 module with ESP32

`headers` are not available:
```
>>> r = urequests.get("https://micropython.org/")
>>> r.status_code
200
>>> r.headers
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'Response' object has no attribute 'headers'
```

The nightly build does not manage to connect to WiFi.
Similar code had no issue with the RP2040 (Pi Pico) port.

I've tried to provide a callback for `parse_headers` to see what's going on but get `TypeError: unexpected keyword argument 'parse_headers'`.

Here are the headers that should be available:
```
$ curl -s -D - -o /dev/null "https://micropython.org/"
HTTP/2 200 
server: nginx/1.10.3
date: Sun, 19 Feb 2023 09:11:34 GMT
content-type: text/html; charset=utf-8
vary: Accept-Encoding
x-frame-options: SAMEORIGIN
strict-transport-security: max-age=15768000
```

**Comment by gnbl** [comment, marked as answer]
It's because these features were added just after the v1.19.1 release:
- https://github.com/micropython/micropython-lib/commit/5854ae1286af2d06e61feea65cfe11a352506dc6
- https://github.com/micropython/micropython-lib/commits/master/python-ecosys/urequests

**Reply by kjm1102** [reply]
Is is necessary to get a post v1.19.1 release to access the improved urequests or can I update just that library?

**Reply by gnbl** [reply]
I assume installing the library using wip would get the latest version and use it instead of any baked-in libraries (I don't know for sure). You could probably also manually install the file or paste the code in your project.
